drivers/lcd/Kconfig

```diff
@@ -1104,6 +1104,13 @@ config LCD_RLANDSCAPE
 		refers one of two orientations where the display is wider than it is
 		tall (LCD_LANDSCAPE is the other).
 
+config LCD_DYN_ORIENTATION
```

Comment:
why not remove LCD_DYN_ORIENTATION and always support dynmaic orientation? Many duplicated code can be removed.

Reply:
To stay compatible with old implementations, and to support better compile time optimization. Duplication is easy to oversee here.

Reply:
But the porotype of st7789_lcdinitialize is changed in this patch, which is an incompatible change. So, I think it's a good chance to let's caller pass in all parameters(offset, resolution and oerientation) instead by Kconfig.
If the compatibility is required, we can:
```
#define st7789_lcdinitialize(spi) st7789_lcdinitialize2(spi, CONFIG_LCD_ST7789_XRES, CONFIG_LCD_ST7789_YRES, CONFIG_LCD_ST7789_XOFFSET, CONFIG_LCD_ST7789_YOFFSET, CONFIG_ST7789_ORIENTATION)
```

Reply:
This define would need to be in the include file to be visible externally. The CONFIG_ST7789_ORIENTATION needs to be created.

I tried to implement this way you suggested the last 20min and came to the conclusion that this is a more complicated less understandabel way.

Nuttx inviolable:
Sometimes Code Duplication is OK